drawBot/context/icnsContext.py

```diff
@@ -0,0 +1,60 @@
+import Quartz
+import os
+import shutil
+import tempfile
+
+from drawBot.misc import executeExternalProcess, DrawBotError
+from drawBot.context.imageContext import ImageContext
+
+
+class ICNSContext(ImageContext):
+
+    fileExtensions = ["icns"]
+
+    allowedPageSizes = [16, 32, 128, 256, 512, 1024]
+
+    def _writeDataToFile(self, data, path, options):
+        # create a iconset folder
+        iconsetPath = tempfile.mkdtemp(suffix=".iconset")
```

Comment:
The rest of the code should be in a try/finally, nw there's a chance this temp folder won't get deleted. Better yet, use `tempfile.TemporaryDirectory()` as a context manager, if Py3-only is ok.

Reply:
we have to setup a plan how long py2 is been supported, it is for now...